File: app/src/main/java/com/hhh/onepiece/main/HomeFragment.java
package com.hhh.onepiece.main;

import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.drawerlayout.widget.DrawerLayout;

import com.google.android.material.tabs.TabLayout;
import com.hhh.mvvm.base.BaseFragment;
import com.hhh.mvvm.multi.MultiFragment;
import com.hhh.mvvm.multi.MultiPagerAdapter;
import com.hhh.onepiece.R;
import com.hhh.onepiece.explore.ExploreFragment;
import com.hhh.onepiece.live.LiveFragment;
import com.hhh.onepiece.subscribe.SubscribeFragment;

public class HomeFragment extends MultiFragment {

  private DrawerLayout mDrawerLayout;

  @Override
  protected int getLayoutResId() {
    return R.layout.home_layout;
  }

  @Override
  protected MultiPagerAdapter onCreatePagerAdapter() {
    return new MultiPagerAdapter(getChildFragmentManager(), R.id.view_pager, 3) {
      @NonNull
      @Override
      protected BaseFragment onCreateFragment(int position) {
        if (position == 0) {
          return new SubscribeFragment();
        } else if (position == 1) {
          return new ExploreFragment();
        } else if (position == 2) {
          return new LiveFragment();
        } else {
          throw new RuntimeException("invalid position!!!");
        }
      }
    };
  }

  @Override
  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    mDrawerLayout = (DrawerLayout) view;
    mViewPager.setCurrentItem(1);
    initBottomBar();
    findViewById(R.id.top_bar_right)
        .setOnClickListener(view1 -> mDrawerLayout.openDrawer(Gravity.RIGHT));
  }

  private void initBottomBar() {
    initTabView(mTabLayout.getTabAt(0), R.drawable.home_tab_icon_subscribe,
        R.string.universal_subscribe);
    initTabView(mTabLayout.getTabAt(1), R.drawable.home_tab_icon_explore, R.string.home_explore);
    initTabView(mTabLayout.getTabAt(2), R.drawable.home_tab_icon_live, R.string.home_live);
  }

  private void initTabView(TabLayout.Tab tab, int drawableId, int textId) {
    tab.setCustomView(R.layout.home_tab_item_layout);
    View view = tab.getCustomView();
    ((ImageView) view.findViewById(R.id.tab_icon)).setImageResource(drawableId);
    ((TextView) view.findViewById(R.id.tab_text)).setText(textId);
  }
}

File: app/src/main/java/com/nb/core/test/main/HomeFragment.kt
package com.nb.core.test.main

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.lifecycle.ViewModelProvider
import com.google.android.material.tabs.TabLayoutMediator
import com.hhh.onepiece.R
import com.nb.core.BaseFragment
import com.nb.core.PagerAdapter
import com.nb.core.PagerFragment
import com.nb.core.test.explore.ExploreFragment
import com.nb.core.test.live.LiveFragment
import com.nb.core.test.subscribe.SubscribeFragment

class HomeFragment : PagerFragment<String>() {

    override val smoothScroll: Boolean
        get() = true
    override fun onCreateViewModel() = ViewModelProvider(this).get(HomeViewModel::class.java)

    override fun onCreatePagerAdapter() = object : PagerAdapter<String>(this) {

        override val isMutable: Boolean
            get() = true

        override fun onCreateFragment(position: Int): BaseFragment {
            return when (position) {
                0 -> SubscribeFragment()
                1 -> ExploreFragment()
                2 -> LiveFragment()
                3 -> ExploreFragment()
                else -> throw RuntimeException("invalid position!!!")
            }
        }

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        pagerView.currentItem = 1
    }

    override fun onCreateTabConfigurationStrategy() =
        TabLayoutMediator.TabConfigurationStrategy { tab, position ->
            tab.setCustomView(R.layout.home_tab_item_layout)
            val imageView: ImageView = tab.customView!!.findViewById(R.id.tab_icon)
            val textView: TextView = tab.customView!!.findViewById(R.id.tab_text)
            when (position) {
                0 -> {
                    imageView.setImageResource(R.drawable.home_tab_icon_subscribe)
                    textView.text = viewModel.getItem(0)
                }
                1 -> {
                    imageView.setImageResource(R.drawable.home_tab_icon_explore)
                    textView.text = viewModel.getItem(1)
                }
                2 -> {
                    imageView.setImageResource(R.drawable.home_tab_icon_live)
                    textView.text = viewModel.getItem(2)
                }
                3 -> {
                    imageView.setImageResource(R.drawable.home_tab_icon_explore)
                    textView.text = viewModel.getItem(3)
                }
            }
        }
}